### PR TITLE
Fix brittle splitting of GUN and target

### LINKED ISSUE #48: List command broken for local registries
```
$ signy --tlscacert=$NOTARY_CA --server https://localhost:4443 sign testdata/cnab/bundle.json localhost:5000/thin-bundle:v1
INFO[0000] Starting to copy image cnab/helloworld:0.1.1 
INFO[0002] Completed image cnab/helloworld:0.1.1 copy   
INFO[0002] Generated relocation map: relocation.ImageRelocationMap{"cnab/helloworld:0.1.1":"localhost:5000/thin-bundle@sha256:a59a4e74d9cc89e4e75dfb2cc7ea5c108e4236ba6231b53081a9e2506d1197b6"} 
INFO[0002] Pushed successfully, with digest "sha256:b4936e42304c184bafc9b06dde9ea1f979129e09a021a8f40abc07f736de9268" 
INFO[0002] Pushed trust data for localhost:5000/thin-bundle:v1: c7e92bd51f059d60b15ad456edf194648997d739f60799b37e08edafd88a81b5 

$ signy --tlscacert=$NOTARY_CA --server https://localhost:4443 verify localhost:5000/thin-bundle:v1
INFO[0000] Pulled trust data for localhost:5000/thin-bundle:v1, with role targets - SHA256: c7e92bd51f059d60b15ad456edf194648997d739f60799b37e08edafd88a81b5 
INFO[0000] Pulling bundle from registry: localhost:5000/thin-bundle:v1 
INFO[0000] Computed SHA: c7e92bd51f059d60b15ad456edf194648997d739f60799b37e08edafd88a81b5 
INFO[0000] The SHA sums are equal: c7e92bd51f059d60b15ad456edf194648997d739f60799b37e08edafd88a81b5

$ signy --tlscacert=$NOTARY_CA --server https://localhost:4443 list localhost:5000/helloworld-thin-bundle   
Error: cannot list targets:localhost:4443 does not have trust data for localhost:5000/helloworld-thin-bundle
```

### LINKED ISSUE #49: Support multiple path separators in bundle names
`localhost:5000/thin-bundle:v1` works, but not `localhost:5000/cnab/thin-bundle:v1`

### LINKED ISSUE #50: Split GUN and target filename correctly
For example, port numbers (e.g., `localhost:5000/helloworld-bundle:v1`) can throw off Signy:

```
{
  "signed": {
    "_type": "Targets",
    "delegations": {
      "keys": {},
      "roles": []
    },
    "expires": "2023-02-10T18:17:02.946024-05:00",
    "targets": {
      "5000/helloworld-bundle": {
        "hashes": {
          "sha256": "x+kr1R8FnWCxWtRW7fGUZImX1zn2B5mzfgjtr9iKgbU=",
          "sha512": "VgQ/Bcy/5IGpguFN65K5hUPM324iFO3jyDLEfFO5RcVzdSilkHhDIJemRxADrWDSX2AVM8qrjTqZg7BRbxt61g=="
        },
        "length": 501
      },
      "5000/thin-bundle": {
        "hashes": {
          "sha256": "x+kr1R8FnWCxWtRW7fGUZImX1zn2B5mzfgjtr9iKgbU=",
          "sha512": "VgQ/Bcy/5IGpguFN65K5hUPM324iFO3jyDLEfFO5RcVzdSilkHhDIJemRxADrWDSX2AVM8qrjTqZg7BRbxt61g=="
        },
        "length": 501
      }
    },
    "version": 3
  },
  "signatures": [
    {
      "keyid": "8862fb206c8bf700caa44620b76f52084d1c0494565abef9c27517546d55105e",
      "method": "ecdsa",
      "sig": "z+4XjB6gLaU2ozfyb9cjg/xWhMaiN3Wety40UHiwJTZjHwufUK5hALBVW5ZSzJI2iRSA5WIpAblKarXPl0LkLA=="
    }
  ]
}
```

### PR DESCRIPTION
**Pull request by radu-matei**
closes #48
closes #49
closes #50 

TODO:

- [x] test with local registry and port numbers
- [x] test with with multi-path hierarchy structure
- [x] test with Docker Hub and Notary
- [x] test `list` command

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -3,7 +3,7 @@ ORG             := cnabio
 BINDIR          := $(CURDIR)/bin
 GOFLAGS         :=
 GOBUILDTAGS     := osusergo
-LDFLAGS         := -w -s
+LDFLAGS         :=
```
Comment (radu-matei) [inline Makefile]: Remove this change.
(In order to debug a Go binary you need to include DWARF symbols, which are stripped with the flags we use by default to build Signy).

File: pkg/tuf/sign.go [hunk 2]
```diff
@@ -17,16 +15,20 @@ func SignAndPublish(trustDir, trustServer, ref, file, tlscacert, rootKey, timeou
 		return nil, fmt.Errorf("cannot ensure trust directory: %v", err)
 	}
 
-	gun, name := cnab.SplitTargetRef(ref)
+	// gun, name := cnab.SplitTargetRef(ref)
```
Comment (radu-matei) [inline pkg/tuf/sign.go]: Remove this commented line.

File: pkg/tuf/verify.go [hunk 3]
```diff
@@ -76,8 +76,13 @@ func verifyTargetSHAFromBytes(target *client.TargetWithRole, buf []byte) error {
 
 // GetTargetAndSHA returns the target with roles and the SHA256 of the target file
 func GetTargetAndSHA(ref, trustServer, tlscacert, trustDir, timeout string) (*client.TargetWithRole, string, error) {
-	gun, name := cnab.SplitTargetRef(ref)
-	target, err := GetTargetWithRole(gun, name, trustServer, tlscacert, trustDir, timeout)
+	// gun, name := cnab.SplitTargetRef(ref)
```
Comment (radu-matei) [inline pkg/tuf/verify.go]: Remove this commented line.

**Comment by radu-matei** [comment]
- local registry and trust server, with port number in URL:

```
 $ signy 
   --tlscacert=$NOTARY_CA 
   --server https://localhost:4443 
   sign testdata/cnab/bundle.json localhost:5000/thin-bundle:v1

INFO[0000] Starting to copy image cnab/helloworld:0.1.1
INFO[0001] Completed image cnab/helloworld:0.1.1 copy
INFO[0001] Generated relocation map: relocation.ImageRelocationMap{"cnab/helloworld:0.1.1":"localhost:5000/thin-bundle@sha256:a59a4e74d9cc89e4e75dfb2cc7ea5c108e4236ba6231b53081a9e2506d1197b6"}
INFO[0002] Pushed successfully, with digest "sha256:b4936e42304c184bafc9b06dde9ea1f979129e09a021a8f40abc07f736de9268"
INFO[0002] Root key found, using: ac20f61a2b72960d46f3c3bacbfe63ae1064165f477fc343911ebfdca7baf345
INFO[0002] Pushed trust data for localhost:5000/thin-bundle:v1: c7e92bd51f059d60b15ad456edf194648997d739f60799b37e08edafd88a81b5

 $ signy 
   --tlscacert=$NOTARY_CA 
   --server https://localhost:4443 
   verify localhost:5000/thin-bundle:v1
INFO[0000] Pulled trust data for localhost:5000/thin-bundle:v1, with role targets - SHA256: c7e92bd51f059d60b15ad456edf194648997d739f60799b37e08edafd88a81b5
INFO[0000] Pulling bundle from registry: localhost:5000/thin-bundle:v1
INFO[0000] Computed SHA: c7e92bd51f059d60b15ad456edf194648997d739f60799b37e08edafd88a81b5
INFO[0000] The SHA sums are equal: c7e92bd51f059d60b15ad456edf194648997d739f60799b37e08edafd88a81b5

$ signy 
   --tlscacert=$NOTARY_CA 
   --server https://localhost:4443 
   list localhost:5000/thin-bundle
v1      c7e92bd51f059d60b15ad456edf194648997d739f60799b37e08edafd88a81b5
```

- local registry and trust server, with port number, and multiple paths:

```
 $ signy 
   --tlscacert=$NOTARY_CA 
   --server https://localhost:4443 
   sign testdata/cnab/bundle.json localhost:5000/multi-path-repo/thin-bundle:v1

INFO[0000] Starting to copy image cnab/helloworld:0.1.1
INFO[0001] Completed image cnab/helloworld:0.1.1 copy
INFO[0001] Generated relocation map: relocation.ImageRelocationMap{"cnab/helloworld:0.1.1":"localhost:5000/multi-path-repo/thin-bundle@sha256:a59a4e74d9cc89e4e75dfb2cc7ea5c108e4236ba6231b53081a9e2506d1197b6"}
INFO[0001] Pushed successfully, with digest "sha256:b4936e42304c184bafc9b06dde9ea1f979129e09a021a8f40abc07f736de9268"
INFO[0001] Root key found, using: ac20f61a2b72960d46f3c3bacbfe63ae1064165f477fc343911ebfdca7baf345
INFO[0001] Pushed trust data for localhost:5000/multi-path-repo/thin-bundle:v1: c7e92bd51f059d60b15ad456edf194648997d739f60799b37e08edafd88a81b5

$ signy 
   --tlscacert=$NOTARY_CA 
   --server https://localhost:4443 
   verify localhost:5000/multi-path-repo/thin-bundle:v1

INFO[0000] Pulled trust data for localhost:5000/multi-path-repo/thin-bundle:v1, with role targets - SHA256: c7e92bd51f059d60b15ad456edf194648997d739f60799b37e08edafd88a81b5
INFO[0000] Pulling bundle from registry: localhost:5000/multi-path-repo/thin-bundle:v1
INFO[0000] Computed SHA: c7e92bd51f059d60b15ad456edf194648997d739f60799b37e08edafd88a81b5
INFO[0000] The SHA sums are equal: c7e92bd51f059d60b15ad456edf194648997d739f60799b37e08edafd88a81b5

$ signy 
   --tlscacert=$NOTARY_CA 
   --server https://localhost:4443 
   list localhost:5000/multi-path-repo/thin-bundle

v1      c7e92bd51f059d60b15ad456edf194648997d739f60799b37e08edafd88a81b5
```


- Docker Hub and Notary:

```
$ signy sign testdata/cnab/bundle.json radumatei/test-signy-ref:v1

INFO[0000] Starting to copy image cnab/helloworld:0.1.1
INFO[0002] Completed image cnab/helloworld:0.1.1 copy
INFO[0002] Generated relocation map: relocation.ImageRelocationMap{"cnab/helloworld:0.1.1":"docker.io/radumatei/test-signy-ref@sha256:a59a4e74d9cc89e4e75dfb2cc7ea5c108e4236ba6231b53081a9e2506d1197b6"}
INFO[0004] Pushed successfully, with digest "sha256:bc9ee7c4f564aaa22ece941e9188186da270e872074d7475b1db8f8b9364c212"
INFO[0005] Root key found, using: ac20f61a2b72960d46f3c3bacbfe63ae1064165f477fc343911ebfdca7baf345
INFO[0006] Pushed trust data for radumatei/test-signy-ref:v1: c7e92bd51f059d60b15ad456edf194648997d739f60799b37e08edafd88a81b5

$ signy verify radumatei/test-signy-ref:v1

INFO[0001] Pulled trust data for radumatei/test-signy-ref:v1, with role targets - SHA256: c7e92bd51f059d60b15ad456edf194648997d739f60799b37e08edafd88a81b5
INFO[0001] Pulling bundle from registry: radumatei/test-signy-ref:v1
INFO[0002] Computed SHA: c7e92bd51f059d60b15ad456edf194648997d739f60799b37e08edafd88a81b5
INFO[0002] The SHA sums are equal: c7e92bd51f059d60b15ad456edf194648997d739f60799b37e08edafd88a81b5

$ signy list radumatei/test-signy-ref

Error: cannot list targets:notary.docker.io does not have trust data for radumatei/test-signy-ref

$ signy list docker.io/radumatei/test-signy-ref

v1      c7e92bd51f059d60b15ad456edf194648997d739f60799b37e08edafd88a81b5
```

Notice that the list command for Docker Hub still requires you to prepend `docker.io` to the username and repository.

**Review by trishankatdatadog — COMMENTED** [review]
Awesome! Would you please add some unit tests for this?

**Comment by radu-matei** [comment]
Added a test, PTAL.

**Review by trishankatdatadog — APPROVED** [review]
💯

**Pull request merged**